src/Contacts/TSThread.m

```diff
@@ -19,8 +19,10 @@ @interface TSThread ()
 @property (nonatomic, retain) NSDate *creationDate;
 @property (nonatomic, copy) NSDate *archivalDate;
 @property (nonatomic, retain) NSDate *lastMessageDate;
-@property (nonatomic, copy) NSString *latestMessageId;
 @property (nonatomic, copy) NSString *messageDraft;
+
+- (TSInteraction *) lastInteraction;
+
 @end
```

Comment:
I'm not storing another serialization of the interaction - take a peak at the implementation a few lines down.

Essentially, I'm just taking advantage of our existing YapDatabase view which *already* keeps track of lastObjectInGroup, vs us redundantly (and incorrectly) reimplementing some of that logic via our "lastMessageId" field.